packages/backend/src/util/handlers/createExpressSynthqlHandler.ts

```diff
@@ -0,0 +1,31 @@
+import { QueryEngine } from '../..';
+import { Request, Response } from 'express';
+
+type ExpressSynthqlHandler = (req: Request, res: Response) => void;
+
+export async function createExpressSynthqlHandler<T>(
+    queryEngine: QueryEngine<T>,
+): Promise<ExpressSynthqlHandler> {
+    return async (req, res) => {
+        const headers = req.headers;
+
+        const query = JSON.parse(req.body);
+
+        res.status(200);
+        res.setHeader('Content-Type', 'application/x-ndjson');
+
+        try {
+            for await (const intermediateResult of queryEngine.execute(query, {
+                returnLastOnly: headers['x-return-last-only'] === 'true',
+            })) {
+                res.write(JSON.stringify(intermediateResult) + '\n');
```

Comment:
Concatenating ad this level is creating an unnecesay copy of the entire JSON string.

Write the lines separately e.g. res.write(json); res.write('\n')

Reply:
Understood. On it!